jetstream/pages-ranking-experiment-revised.toml

```diff
@@ -1,18 +1,88 @@
 [metrics]
-weekly = ["history_clicks", "bookmark_clicks", "search_suggestion_clicks"]
-overall = ["history_clicks", "bookmark_clicks", "search_suggestion_clicks"]
+weekly = ["history_impressions", "bookmark_impressions", "search_suggestion_impressions",
+          "history_clicks", "bookmark_clicks", "search_suggestion_clicks"]
+overall = ["history_impressions", "bookmark_impressions", "search_suggestion_impressions",
+           "history_clicks", "bookmark_clicks", "search_suggestion_clicks"]
+
+
+[metrics.history_impressions]
+select_expression = "SUM(IF(product_result_type = 'history', num_impressions, 0))"
+data_source = "urlbar_events_impressions_by_product_result_type"
+[metrics.history_impressions.statistics.bootstrap_mean]
+
+[metrics.bookmark_impressions]
+select_expression = "SUM(IF(product_result_type = 'bookmark', num_impressions, 0))"
+data_source = "urlbar_events_impressions_by_product_result_type"
+[metrics.bookmark_impressions.statistics.bootstrap_mean]
+
+[metrics.search_suggestion_impressions]
+select_expression = "SUM(IF(product_result_type = 'default_partner_search_suggestion', num_impressions, 0))"
+data_source = "urlbar_events_impressions_by_product_result_type"
+[metrics.search_suggestion_impressions.statistics.bootstrap_mean]
 
 [metrics.history_clicks]
 select_expression = "COUNT(DISTINCT(CASE WHEN is_terminal and event_action = 'engaged' and product_engaged_result_type = 'history' THEN event_id ELSE NULL END))"
 data_source = "urlbar_events"
 [metrics.history_clicks.statistics.bootstrap_mean]
 
+[metrics.bookmark_clicks]
+select_expression = "COUNT(DISTINCT(CASE WHEN is_terminal and product_engaged_result_type = 'bookmark' THEN event_id ELSE NULL END))"
```

Comment:
this should have `event_action = 'engaged'` as well to count clicks. However, it can be a `COUNTIF` rather than `COUNT DISTINCT` using the `urlbar_events` data source